internal/controllers/bundledeployment/bundledeployment.go

```diff
@@ -449,27 +489,35 @@ const (
 	stateError        releaseState = "Error"
 )
 
-func (c *controller) getReleaseState(cl helmclient.ActionInterface, bd *rukpakv1alpha2.BundleDeployment, chrt *chart.Chart, values chartutil.Values, post *postrenderer) (*release.Release, releaseState, error) {
+// TODO: update this to return desired release and run dry runs for install and uninstall
+func (c *controller) getReleaseState(cl helmclient.ActionInterface, bd *rukpakv1alpha2.BundleDeployment, chrt *chart.Chart, values chartutil.Values, post *postrenderer) (*release.Release, *release.Release, releaseState, error) {
 	currentRelease, err := cl.Get(bd.GetName())
 	if err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
-		return nil, stateError, err
+		return nil, nil, stateError, err
 	}
 	if errors.Is(err, driver.ErrReleaseNotFound) {
-		return nil, stateNeedsInstall, nil
+		desiredRelease, err := cl.Install(bd.GetName(), bd.Spec.InstallNamespace, chrt, values, func(i *action.Install) error {
+			i.DryRun = true
+			return nil
+		}, helmclient.AppendInstallPostRenderer(post))
+		if err != nil {
+			return nil, nil, stateError, err
+		}
+		return nil, desiredRelease, stateNeedsInstall, nil
 	}
 	desiredRelease, err := cl.Upgrade(bd.GetName(), bd.Spec.InstallNamespace, chrt, values, func(upgrade *action.Upgrade) error {
 		upgrade.DryRun = true
 		return nil
 	}, helmclient.AppendUpgradePostRenderer(post))
 	if err != nil {
-		return currentRelease, stateError, err
+		return currentRelease, nil, stateError, err
 	}
 	if desiredRelease.Manifest != currentRelease.Manifest ||
 		currentRelease.Info.Status == release.StatusFailed ||
 		currentRelease.Info.Status == release.StatusSuperseded {
-		return currentRelease, stateNeedsUpgrade, nil
+		return currentRelease, desiredRelease, stateNeedsUpgrade, nil
 	}
-	return currentRelease, stateUnchanged, nil
+	return currentRelease, nil, stateUnchanged, nil
```

Comment:
Seems like it might make sense to return `desiredRelease` even in line 520. Is there a good reason not to?

And looking at 515-518, I wonder if we should do something like this instead:
```suggestion
	relState := stateUnchanged
	if desiredRelease.Manifest != currentRelease.Manifest ||
 		currentRelease.Info.Status == release.StatusFailed ||
 		currentRelease.Info.Status == release.StatusSuperseded {
 		relState = stateNeedsUpgrade
 	}
 	return currentRelease, desiredRelease, relState, nil
```

Reply:
>Seems like it might make sense to return desiredRelease even in line 520. Is there a good reason not to?

No particular reason. I was just trying to keep the impact to the logic pretty minimal and didn't think it would be worth returning the desired release in the case where it wouldn't be used, but that is probably better left to the caller to decide than the function. Will tweak this a bit.

Reply:
updated in the latest push